lnurl-allowance.md

```diff
@@ -0,0 +1,197 @@
+# LNURL-allowance
+
+The LNURL Allowance scheme allows a `LN SERVICE` and a `LN WALLET` to establish a `pairing` over a WebSocket connection. The idea is that `LN SERVICE` will provide an LNURL that returns some parameters (including a socket URL) to `LN WALLET`. `LN WALLET` will diplay a pairing dialog and once accepted by the user, `LN WALLET` will start a socket connection to `LN SERVICE`. This will allow for `LN WALLET` to pay/receive funds to/from `LN SERVICE` without user interaction or QR Code scanning, up to the established allowance amount.
+
+## Scan QR code to establish allowance/pairing
+
+### WALLET to SERVICE interaction flow:
+
+1. User scans a LNURL QR code or accesses a `lightning:LNURL..` link with `LN WALLET` and `LN WALLET` decodes LNURL.
+
+2. `LN WALLET` makes a GET request to `LN SERVICE` using the decoded LNURL.
+
+3. `LN WALLET` gets JSON response from `LN SERVICE` of form:
+
+```
+{
+    socket: String, // the URL from LN SERVICE which will accept the WebSocket connections from the WALLET
+    k1: String, // random or non-random string to identify the WALLET when receiving connection to socket
+    recommendedAllowanceAmount: Millsatoshis, // amount of satoshis the SERVICE recommends WALLET to ask user for this allowance/pairing
+    image: Base64, // base64-encoded LN SERVICE image for LN WALLET to display to user when establishing connection
```

Comment:
What about the image mimetype? I think sometimes it is necessary to know that beforehand.

Reply:
Yeah good point. Should we enforce a mimetype, or do we want to add another property for it? At that point, we may want to follow the approach for the multi-property `content metadata` property present in LNURL Pay?

Reply:
I don't know. That property is only there in that specific format because it needed to be hashed and included in the invoice description. There are nicer ways to do it without that constraint.

Anyway, I'm implementing this assuming images will be PNG, but I think this only matters for Telegram, and we can add an optional parameter later.